Makefile

```diff
@@ -11,37 +11,53 @@ BUILD_INFO=-ldflags "-X $(BUILD_INFO_IMPORT_PATH).Version=$(VERSION)"
 COMP_REL_PATH=internal/components/components.go
 MOD_NAME=github.com/open-telemetry/opentelemetry-collector-contrib
 
-# ALL_MODULES includes ./* dirs (excludes . dir and example with go code)
-ALL_MODULES := $(shell find . -type f -not -path '*/pkg/stanza/*' -name "go.mod" -exec dirname {} \; | sort | egrep  '^./' )
-
-# Modules to run integration tests on.
-# XXX: Find a way to automatically populate this. Too slow to run across all modules when there are just a few.
-INTEGRATION_TEST_MODULES := \
-	internal/containertest \
-	receiver/apachereceiver \
-	receiver/dockerstatsreceiver \
-	receiver/jmxreceiver/ \
-	receiver/kafkametricsreceiver \
-	receiver/memcachedreceiver \
-	receiver/mysqlreceiver \
-	receiver/nginxreceiver \
-	receiver/postgresqlreceiver \
-	receiver/redisreceiver \
-	receiver/riakreceiver \
-	receiver/zookeeperreceiver \
-	extension/observer/dockerobserver
+GROUP ?= all
+FOR_GROUP_TARGET=for-$(GROUP)-target
+
+TEMP_EX_STANZA=-not -path "./pkg/stanza/*"
+FIND_MOD_ARGS=-type f -name "go.mod" $(TEMP_EX_STANZA)
+TO_MOD_DIR=dirname {} \; | sort | egrep  '^./'
+EX_COMPONENTS=-not -path "./receiver/*" -not -path "./processor/*" -not -path "./exporter/*" -not -path "./extension/*"
+EX_INTERNAL=-not -path "./internal/*"
+
+# NONROOT_MODS includes ./* dirs (excludes . dir)
+NONROOT_MODS := $(shell find . $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+
+RECEIVER_MODS_0 := $(shell find ./receiver/[a-k]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+RECEIVER_MODS_1 := $(shell find ./receiver/[l-z]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+RECEIVER_MODS := $(RECEIVER_MODS_0) $(RECEIVER_MODS_1)
+PROCESSOR_MODS := $(shell find ./processor/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+EXPORTER_MODS := $(shell find ./exporter/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+EXTENSION_MODS := $(shell find ./extension/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+INTERNAL_MODS := $(shell find ./internal/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
+OTHER_MODS := $(shell find . $(EX_COMPONENTS) $(EX_INTERNAL) $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) ) $(PWD)
+ALL_MODS := $(RECEIVER_MODS) $(PROCESSOR_MODS) $(EXPORTER_MODS) $(EXTENSION_MODS) $(INTERNAL_MODS) $(OTHER_MODS)
```

Comment:
Would it make sense to spilt processors and/or exporters and/or extensions into multiple groups as well?  I know there aren't as many of those as there are receivers right now, but I'd expect them to continue to grow. Would be great for the CI to be able to handle increases in scale without having to come back and update this process again.

Reply:
In my opinion we're better off parallelizing only to the extent that it benefits the current set of components. If we end up seeing that splitting another group would be helpful, it's it fairly easy thing to do, given the pattern established here.

Reply:
Is there a cost to the parallelization?